### PR TITLE
Improve check for bbox

### PR DESCRIPTION
**Pull request by oscargus**
## PR summary

Passing something else than a `Transform` would lead to an `AttributeError`, but now the better error message should show up.

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/transforms.py [hunk 1]
```diff
@@ -1101,7 +1101,7 @@ def __init__(self, bbox, transform, **kwargs):
         bbox : `Bbox`
         transform : `Transform`
         """
-        if not bbox.is_bbox:
+        if not hasattr(bbox, 'is_bbox') or not bbox.is_bbox:
```
Comment (QuLogic) [inline lib/matplotlib/transforms.py:1104]: You could do this with `getattr`, I think?
```suggestion
        if not getattr(bbox, 'is_bbox', False):
```

Comment (oscargus) [reply lib/matplotlib/transforms.py:1104]: Ahh, clever! (A bit too tired, so trying to do "simple" things, but they do not always turn out that good...)

**Review by QuLogic — APPROVED** [review]

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Doc build failed?

**Comment by QuLogic** [comment]
Yes, everything failed earlier this morning due to network issues. I've restarted them all.

**Review by anntzer — CHANGES_REQUESTED** [review]
Seems more natural to just check_isinstance(BboxBase, bbox=bbox) if we want to go that direction, no?  (Also consistent with the line immediately after.)
Feel free to dismiss if you disagree.

**Comment by oscargus** [comment]
> Seems more natural to just check_isinstance(BboxBase, bbox=bbox) if we want to go that direction, no? (Also consistent with the line immediately after.) Feel free to dismiss if you disagree.

Except that a TypeError will be raised. Which on the other hand makes more sense. I can change if there is agreement on this.

**Comment by oscargus** [comment]
I added the change as a second commit. Not sure if I should deprecate it as a property or classproperty though.

File: lib/matplotlib/transforms.py [hunk 2]
```diff
@@ -2547,8 +2544,8 @@ def __init__(self, boxin, boxout, **kwargs):
         Create a new `BboxTransform` that linearly transforms
         points from *boxin* to *boxout*.
         """
-        if not boxin.is_bbox or not boxout.is_bbox:
-            raise ValueError("'boxin' and 'boxout' must be bbox")
+        _api.check_isinstance(BboxBase, boxin=boxin)
+        _api.check_isinstance(BboxBase, boxout=boxout)
```
Comment (QuLogic) [inline lib/matplotlib/transforms.py:2548]:
```suggestion
        _api.check_isinstance(BboxBase, boxin=boxin, boxout=boxout)
```

**Review by timhoffm — APPROVED** [review]

**Pull request merged**